main.go

```diff
@@ -47,29 +55,60 @@ func runMain() error {
 		return err
 	}
 
-	camera := lepton3.New(conf.SPISpeed)
-	err = camera.Open()
-	if err != nil {
-		return err
+	var camera *lepton3.Lepton3
+	defer func() {
+		if camera != nil {
+			camera.Close()
+		}
+	}()
```

Comment:
Nothing ever sets `camera` to `nil` once it's created so it's fine to just do `defer camera.Close()`.